### ISSUE TITLE
Use of `titlematch` causes Notice: Trying to access array offset on value of type null

### ISSUE THREAD

**Issue by designnerd** [Universal-Omega/DynamicPageList3#19]
Anytime `titlematch` is used in 1.35 (1.35.0) I receive a bit of a mysterious php on tons of pages on my site and they pile up in the php logs:
Notice: Trying to access array offset on value of type null in /home/username/site.com/w/extensions/DynamicPageList3/includes/Parameters.php on line 853

E.g., when using any of the following:

```
{{#dpl: 
|randomcount=1
|escapelinks=false
|namespace=File
|category=Images:{{{category|Promotions}}}
|titlematch={{{titlematch|}}}
|listseparators=,[[%PAGE%|250px|{{#if:{{{link|}}} |{{!}} link={{{link}}} | link=[[:Category:Images:{{{category|Promotions}}}]] }}]]\n,,
}}
```

```
{{#dpl: 
|category=Images
|titlematch=A%
}}
```

Since `titlematch` is used a **ton** on our site in different places for different uses, so this is going to fill up our logs rather fast. Is there a way to disable this? I ask because `titlematch` is working as expected, so I'm not really even sure what this notice means or how it might be useful. 

Thanks!

**Comment by Universal-Omega** [comment]
It looks like you don't have the `title` parameter set? https://github.com/Universal-Omega/DynamicPageList3/blob/169b2009942abcc849d930246199565d75df5d02/includes/Parameters.php#L852-L853

**Comment by designnerd** [comment]
The second example has a value for |titlematch=a% straight from the manual and it kicks the same error.

**Comment by designnerd** [comment]
Ah, I see what's going on. 
For known pages it only kicks the error on page preview:
https://www.dragon-mania-legends.info/wiki/User:CrystalClear/Sandbox

When a variable is used (which is for parser functions version) it kicks on any page the variable is then used on but only in 1.35 (doesn't do it on our 1.33 site). 
https://www.dragon-mania-legends.info/wiki/Category:Events
Which uses the template with a parser function in titlematch again:
https://www.dragon-mania-legends.info/wiki/Template:TableOfEventsPerYear

I mean, I guess it is safe to ignore it as long as it is working and we set up rotating logs?

**Comment by Universal-Omega** [comment]
> Ah, I see what's going on.
> For known pages it only kicks the error on page preview:
> https://www.dragon-mania-legends.info/wiki/User:CrystalClear/Sandbox
> 
> When a variable is used (which is for parser functions version) it kicks on any page the variable is then used on but only in 1.35 (doesn't do it on our 1.33 site).
> https://www.dragon-mania-legends.info/wiki/Category:Events
> Which uses the template with a parser function in titlematch again:
> https://www.dragon-mania-legends.info/wiki/Template:TableOfEventsPerYear
> 
> I mean, I guess it is safe to ignore it as long as it is working and we set up rotating logs?

It should be safe to ignore, yes but nonetheless I still can hopefully fix this as it's not ideal to fill up logs with this.

**Comment by designnerd** [comment]
Great, thanks for confirming. :) I'm actually going to remove my site links now as that test site should normally be inaccessible.

**Comment by Universal-Omega** [comment]
Looking into this, it is possible to suppress the PHP warning by adding "`@`" to the relevant line of code, which I may do, since it obviously still works and therefore the warning should not matter, but still, it's obviously giving it for a reason so this is bad practice and a last resort.

**Comment by designnerd** [comment]
Hmm, okay, no worries. We'll also discuss if we want to suppress in the meantime using the method you mention--thanks for that! Can I ask you to please remove my URLs from the quote above? Sorry for being a pest, I should have thought it out better if I wanted that testing site URL posted, before I posted it. Apologies, and thanks! :octocat:

**Comment by Universal-Omega** [comment]
Should be fixed with #23.

**Comment by designnerd** [comment]
Apologies for the long delayed reply (had stopped testing 1.35 for quite a while), yes, thank you, this fixed the error. I cross-linked for another user to see who has the same one. I just updated to 1.35.2 and tested and this new works fine without any more errors related to this in the logs.
